app/src/main/java/com/blink22/android/gitdb/GitReposListSearchFragment.java

```diff
@@ -0,0 +1,134 @@
+package com.blink22.android.gitdb;
+
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v4.app.Fragment;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.support.v7.widget.SearchView;
+import android.view.LayoutInflater;
+import android.view.Menu;
+import android.view.MenuInflater;
+import android.view.MenuItem;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Toast;
+
+import java.util.List;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import io.realm.Realm;
+import io.realm.RealmResults;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+public class GitReposListSearchFragment extends Fragment {
+    private static final String DEFAULT_SEARCH_QUERY = "android";
+    @BindView(R.id.repos_recycler_view)
+    RecyclerView mReposRecyclerView;
+    private RepoAdapter mRepoAdapter;
+    private Call<ReposSearchResult> mSearchReposCall;
+    private Realm mRealm;
+
+    public static GitReposListSearchFragment newInstance() {
+        return new GitReposListSearchFragment();
+    }
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setHasOptionsMenu(true);
+        setRetainInstance(true);
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
```

Comment:
@YusufHussein I vote for breaking the method into calls to private methods grouping related stuff.